### ISSUE TITLE
ustream.lxc packer integration

### ISSUE THREAD

**Issue by grenzr** [ustream/packer-builder-lxc#11]
Great work on this builder :) I am successfully building LXC boxes using it, but the major missing part is the ustream.lxc publisher.

I just wanted to find out whether there has been any dev effort on this yet, or can I help collaborate on it as I'd really like to see this feature implemented.

**Comment by oker1** [comment]
Hi, great to hear you are having success with it! We've been using it with a patched version of packer to publish vagrant-lxc boxes: https://github.com/oker1/packer/commits/vagrant-lxc

I didn't send a PR from it because the vagrant-lxc box format was not final at that time. I haven't been following the project since, don't know if it has been finalized or not.

**Comment by grenzr** [comment]
Ah thats great thanks, I'll give it a try. I would consider merging packer 0.7.5 with this vagrant-lxc branch and then just file a PR at mitchellh/packer anyway and get the conversation rolling with that PR? Would be great to get it finally included. Thanks again!

**Comment by oker1** [comment]
That would be great!

**Comment by grenzr** [comment]
You wanna do the PR, or shall I?

**Comment by oker1** [comment]
Go ahead and create it please :)

**Comment by perj** [comment]
Any chance of getting the user variable expansion in first? Or do you want me to do some more work on it?

**Comment by grenzr** [comment]
I've created a fork : https://github.com/grenzr/packer ready for PR (just testing it, and assessing what tests need to be written)
@perj the variable expansion thing is in this builder repo, not packer itself isn't it?

**Comment by fgrehm** [comment]
As per vagrant-lxc box format, I can say that is "final" these days. "Final" as in we haven't had a need to change it in a looooong time ;-)

**Comment by grenzr** [comment]
@fgrehm thats good news, thanks :)

**Comment by perj** [comment]
@grenzr yeah it's a PR here in this repo. I wasn't aware of the other one.

**Comment by oker1** [comment]
@perj sorry I couldn't get around to test it properly before merging it. @grenzr are you interested in testing it? If you vouch for it I'd be happy to merge it.

**Comment by grenzr** [comment]
Hi all - sorry have been away for a while. I have successfully tested it and it works fine. I was just wondering whether any tests needed to be written for it before submission to main packer repo. I will hopefully get a chance to look at this again this week if nobody else has already?

**Comment by harlanbarnes** [comment]
@grenzr in answer to your question to @perj above, the User variable stuff is in packer-builder-lxc. It's issue #2 and has a pr #7. I agree with @perj that this is probably a necessary change before it gets merged into packer proper (since the expectation for all other packer resources is to handle the User variables.)

**Comment by cstyan** [comment]
Any word on when the user variables PR might be merged?

**Comment by oker1** [comment]
I've merged it now. I couldn't test it extensively but seems like it didn't break anything.

**Comment by cstyan** [comment]
I haven't run into any problems yet, but I'm going to be using this builder frequently so I'll let you know if I run into any issues regarding the variables.

**Comment by perj** [comment]
Guess I should leave a comment to point out that I'm still around. :smile:

**Comment by oker1** [comment]
I've released a new preview version, which creates a working vagrant box for the latest vagrant-lxc release. The example config uses the compress postprocessor to produce the vagrant box. This can work while we figure out how to create a proper vagrant postprocessor plugin (see https://github.com/mitchellh/packer/pull/2537)
